### ISSUE TITLE
Setup command for contest [cpp]

### ISSUE THREAD

**Issue by ankushbhardwxj** [ankingcodes/codemon#12]
`codemon init <contestName>` should create directories named `A, B, C, D, E, F`, where each directory should contain a `.cpp` file for writing code, `.in` file for inputs for the particular problem, `.out` for the outputs of that problem. The directory structure is as follows: 
For a contest, for example codeforces round 1353, 
```
1353/
   A/
     a.cpp
     a.op
     a.in
  B/
  C/
  ...
  test_case.cpp (program to write testcase) 
  prog 
  test_case (generate testcases in this file)
```
Make sure `.cpp` files created uses template from `~/.codemon` directory at root.
If template doesn't exist, add default template: 
```cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
  return 0;
}
```

**Comment by lucifer0987** [comment]
Hey,
I came to know about this project from NJACK winter of code.
I want to work on this issue. 
Can you please explain about these files? I understood the remaining things.

test_case.cpp (program to write testcase) 
prog 
test_case (generate testcases in this file)

Hoping for a reply soon!!

**Comment by ankushbhardwxj** [comment]
@lucifer0987 - Watch this [video](https://www.youtube.com/watch?v=JXTVOyQpSGM) by Errichto. Note the part where he writes a program to generate testcases. Therefore, `test_case.cpp` is the file where user is supposed to write that program. 
When we compile C++, we get `a.out` which has been renamed as `prog` here. `test_case` is the file where the testcases generated by the program in `test_case.cpp` has to be saved. 

**Note:** Start working only after Dec 1 in order to get points.

**Comment by lucifer0987** [comment]
ok, got it.
Thanks for quick reply.

Just a suggestion.
Since, testcases that anyone is going to generate differs from task to task. Can't we generate these 3 files for each task separately?
That way, we can write the generated testcases directly into the "input" file of each task.

And, each task will have 5 files. 
a.cpp - c++ code
a.in - input given to the code
a.op - output generated by the code
prog (a.out)
test_case.cpp - program for generating testcases

when we run "test_case.cpp", the generated test cases(outputs) are stored in "a.in" directly. From there, we can use the generated test_cases directly.

**Comment by ankushbhardwxj** [comment]
@lucifer0987 - During a contest, we need not write testcases for each problem. If we create 3 files for each task, we will have a lot of redundant files which will simply eat our storage. Therefore, this solution would not be very helpful for some people.
However, if you can add a command `codemon clean`, which clears up all the mess(prog, input.txt, testcase files) after a contest, then this might work.

**Comment by lucifer0987** [comment]
Yeah sure!! 
That would be a good thing.
I will implement codemon clean.
Only the cpp files of the tasks will be remaining.

**Comment by ankushbhardwxj** [comment]
> Only the cpp files of the tasks will be remaining.

There is a catch to this. We can't randomly delete the *mess*. We can directly delete prog. But for `input.txt`, we must check whether it contains something, if its empty delete it else give an error message saying "Input file is not empty, do you want to delete ? (Y/N)` and user can decide what to do with it. For the testcase files, we have to "diff" them before deleting, i.e., if the testcase files had no changes in it then it can be deleted else if any kind of changes was done, they must stay.

**Comment by ankushbhardwxj** [comment]
Can you create a separate issue for `codemon clean` and express these ideas properly there?

**Comment by lucifer0987** [comment]
yeah sure!!
I will make a separate issue for codemon clean. And, meanwhile, work on this issue.

PS. :- can we connect with you anywhere apart from github?

**Comment by ankushbhardwxj** [comment]
> can we connect with you anywhere apart from github?

I've created a gitter chatroom for this project. Check README.

**Comment by ankushbhardwxj** [comment]
@yashshah14093 @lucifer0987 - if you're working on this issue, make sure to update your branch to changes made in #41
